docs/architecture/adr-007.md

```diff
@@ -10,88 +13,101 @@ The current way to define those events is not flexible: events are hardcoded, pa
 
 This can lead to potential issue in the long run when commands needs to be updated to add more parameters (ibc-go/ gravity bridge) or even if we need to implement more "actions". Since contracts are immutable, contract developers won't be able to have access to these functionalities after deploying their CRC20 contracts if we perform an upgrade.
 
-In this ADR, we define generic events format that can be use by smart contract developer to call any "action" upon successful contract execution. 
+In this ADR, we define generic event argument that can be extended to accept new parameters in the future. 
 
 ## Decision
 
-We define 2 specifics events that smart contract can use to trigger specific action after contract execution.
+### __CronosSendToChainV2 Event
 
-```
-# Trigger an action after contract execution which only involve the sender account
-__CosmosCall(string functionName, address sender,  byte[]args)
+#### Changes
 
-# Trigger an action after contract execution that require moving asset from the smart contract "account"
-__CosmosCallAfterPayment(string functionName, address payer, uint amount, byte[]args)
+- Introduce event `__CronosSendToChainV2`, successor of event `__CronosSendToChain`;
+- Add new argument `string channelId` to accept destination channel Id;
```

Comment:
what the use of channelID?

I think its better to separate CronosSendToIBC and CronosSendToChain, otherwise we would have the same security concern as the previous proposed ADR

Reply:
Actually it's a typo. This is the `__CronosSendToIBC` event